dask_cuda/disk_io.py

```diff
@@ -201,18 +206,25 @@ def disk_read(header: Mapping, gds=False) -> list:
     """
     ret = []
     if gds:
-        import cucim.clara.filesystem as cucim_fs  # isort:skip
+        import kvikio  # isort:skip
 
-        with cucim_fs.open(header["path"], "rb") as f:
+        with kvikio.CuFile(header["path"], "rb") as f:
             file_offset = 0
+            futs = []
             for length, is_cuda in zip(header["frame-lengths"], header["cuda-frames"]):
                 if is_cuda:
                     buf = get_new_cuda_buffer()(length)
                 else:
                     buf = np.empty((length,), dtype="u1")
-                f.pread(buf=buf, count=length, file_offset=file_offset, buf_offset=0)
+                futs.append(
+                    f.pread(
+                        buf=buf, count=length, file_offset=file_offset, buf_offset=0
+                    )
+                )
```

Comment:
Maybe some kind of `readline`s type API would help consolidate this? May need more thought. Also raised in issue ( https://github.com/rapidsai/kvikio/issues/77 )